system/Database/Postgre/Connection.php

```diff
@@ -128,11 +128,14 @@ public function getVersion(): string
             return $this->dataCache['version'];
         }
 
-        if (! $this->connID || ($pgVersion = pg_version($this->connID)) === false) {
+        if (! $this->connID) {
             $this->initialize();
         }
 
-        return isset($pgVersion['server']) ? $this->dataCache['version'] = $pgVersion['server'] : false;
+        $pgVersion                  = pg_version($this->connID);
+        $this->dataCache['version'] = $pgVersion['server'] ?? 'PostgreSQL 7.4 or lower';
```

Comment:
`getVersion()` returns the version number like `'12.1'`, and the number should be used with [version_compare()](https://www.php.net/manual/en/function.version-compare.php)

Is it impossible to get the version number 7.4 or lower?

Reply:
> pg_version() returns an array with the client, protocol and server version. Protocol and server versions are only available if PHP was compiled with PostgreSQL 7.4 or later. 
https://www.php.net/manual/en/function.pg-version.php#refsect1-function.pg-version-description

There is no version requirement in the docs:
https://codeigniter4.github.io/CodeIgniter4/intro/requirements.html#supported-databases

But PostgreSQL 11 will stop receiving fixes on November 9, 2023.
https://www.postgresql.org/about/news/postgresql-153-148-1311-1215-and-1120-released-2637/

@codeigniter4/core-team @codeigniter4/database-team 
Do we need to support versions prior to 7.4?
Why don't we add `(version 7.4 and above only)` in the docs?

Reply:
PostgreSQL 7.4 seems too old, it seems it need to be documented in docs

Reply:
> getVersion() returns the version number like '12.1', and the number should be used with [version_compare()](https://www.php.net/manual/en/function.version-compare.php)

that was the question in the related issue #7488

Reply:
If there is no `$pgVersion['server']`, we should return an empty string.

Adding a note `(version 7.4 and above only)` is a good idea.